linetools/lists/linelist.py

```diff
@@ -341,11 +345,16 @@ def set_extra_columns_to_datatable(self, abundance_type='solar', ion_correction=
             Ionization correction. Options are:
                 'none' : No correction applied, so this column will
                          be filled with zeros. (Default)
+         redo : bool, optional
+            Remake the extra columns
 
         Note: This function is only implemented for the following
         lists: HI, ISM, EUV, Strong.
 
         """
+        # Avoid redo (especially for caching)
+        if ('ion_name' in self._data.keys()) and (not redo):
+            return
 
```

Comment:
Yeah, I almost went that route...
I'll leave that refactor to you.  :)

Please merge if you are ok with it.